### ISSUE TITLE
The Service Name generated in the DNS tools does not use the canonical hostname

### ISSUE THREAD

**Issue by sarahhenkens** [arangodb/kube-arangodb#738]
The following utility inside the DNS tools is not generating the fully qualified DNS name of the services  https://github.com/arangodb/kube-arangodb/blob/1a5a2691789477b5d849a47f44ee56f0a966a4e4/pkg/util/k8sutil/dns.go#L53-L56

When using a custom DNS setup and other more advanced systems. The `/etc/resolv.conf` does not contain the `default.svc.cluster.local svc.cluster.local cluster.local` CNAME shortcuts.
```
nameserver 172.17.0.1
search foo.bar.com foo.foobar.com
```

As a result, the operator is unable to reach the pods it creates because it is hardcoded at `<svc-name>.<namespace>.svc`.

The `arangodb-dev.default.svc.cluster.local` full name works as expected while `arangodb-dev.default.svc` forces some default configuration in the clusters in which the operator runs.

### Recommendation

I suggest we modify this utility to always generate the full name:
```go
 // CreateServiceDNSName returns the DNS of a service. 
func CreateServiceDNSName(svc *core.Service) string {
  return fmt.Sprintf("%s.%s.svc.cluster.local", svc.GetName(), svc.GetNamespace())
} 
```

**Comment by ajanikow** [comment]
Hello!

You can use part of this to achieve same thing:
https://github.com/arangodb/kube-arangodb/blob/1a5a2691789477b5d849a47f44ee56f0a966a4e4/pkg/util/k8sutil/dns.go#L58

https://github.com/arangodb/kube-arangodb/blob/1a5a2691789477b5d849a47f44ee56f0a966a4e4/pkg/apis/deployment/v1/deployment_spec.go#L120

https://github.com/arangodb/kube-arangodb/blob/1a5a2691789477b5d849a47f44ee56f0a966a4e4/pkg/deployment/resources/secrets.go#L154 

If you will set  .spec.ClusterDomain to cluster.local all certs gonna be generated with proper SAN DNS entries. (we know about invalid camel case, fix will be in 1.2.0)

Best Regards,
Adam.

**Comment by ajanikow** [comment]
Hello!

I'm back to you after long time but this issue has been fixed. 

Currently, you can change Network method in deployment by changing https://github.com/arangodb/kube-arangodb/blob/master/pkg/apis/deployment/v1/deployment_spec.go#L165

In 1.3.0 we will change default from `headless` to `dns`.

Best Regards,
Adam.
